docs/architecture.md

```diff
@@ -1,14 +1,102 @@
 
-### Missing parts
+To explain the architecture, we will see how a request for data flows
+through cozy-client and how finally the UI is updated.
 
-- Init
-- Associations
-- Mutations
+### Query defitions
+
+To query data, we create a *query definition*: an object describing documents to fetch.
+
+```
+{
+  doctype: "io.cozy.todos",
+  selector: {
+    "finished": true
+  }
+}
+```
+
+It is typically created via the helper `Q` that provides a fluid interface
+to create a *Query Definition*.
+
+```jsx
+const qdef = Q('io.cozy.todos').where({ finished: true })
+```
+
+Then we need to execute this query, fetch the data, and storing it.
+
+```jsx
+await client.query(qdef)
+```
+
+### Links
+
+When executed, *Query definitions* are passed down to *Links*.
+
+Links accept query definitions and can choose
+
+- either to return a response
+- or to pass the query definition down to the next link. 
+
+ℹ️ This is how offline support is implemented : on mobile, a PouchLink is added as a link to cozy-client, before the default StackClientLink. The 
+Pouch link can decide either to pass down the query definition to the next link (if the pouch link has not been synchronized for example), or to respond to the request by communicating directly with Pouch.
+
+At the moment there are only two links
+
+- StackLink : fetches data over HTTP by communicating to the stack via the
+StackClient.
+- PouchLink : fetches data from a local Pouch. Useful to have offline ready
+  applications.
+
+### Store
+
+When links have responded with data, the data is stored inside the
+*redux store* that is internal to cozy-client. This redux store
+brings observability to cozy-client, and allows for connection of
+UI components to the data.
+
+ℹ️ You do not need to use Redux in your application to use cozy-client.
+
+ℹ️ You can connect your own store to cozy-client. It is useful for more
+advanced techniques where you create selectors directly on the data
+of cozy-client.
+
+The redux is composed of two slices, "documents" and "queries". Documents
+stores the data index by doctype then \_id. Queries store information
+for a particular query:
+
+- ids of the documents that match the query
+- whether the server has more documents that can be fetched (useful for
+  pagination)
+- whether the query is being loaded.
+
+```
+{
+    documents: {
+        'io.cozy.todos': {
+            'todo-id-1': {
+                finished: true,
+                label: "Add architecture document for cozy-clietn"
+            }
+        }
+    },
+    queries: {
+        finishedTodos: {
+            data: ['todo-id-1']
+        }
+    }
+
+}
+```
+
+The glue between the Redux store and the UI is done via ObservableQuery.
+ObservableQuery are objects instantiated by a <Query /> component. Their
+role is to react to store changes and wake the component. They should not
+be used directly.
```

Comment:
> They should not be used directly

What should be used in place of?

Reply:
Nothing, Query and useQuery do the job for you. I am adding the mention, thanks.